astropy/io/fits/hdu/hdulist.py

```diff
@@ -689,6 +698,13 @@ def writeto(self, fileobj, output_verify='exception', clobber=False,
             hdu._prewriteto(checksum=checksum)
             try:
                 hdu._writeto(hdulist._file)
+            except IOError, e:
+                free_space = get_free_space_in_dir(os.getcwd())
+                hdulist_size = np.sum(hdu.size for hdu in self)
+                if not fileobj.file_like and free_space < hdulist_size:
+                    raise IOError("Not enough space on disk. " + str(e))
+                else:
+                    raise IOError(str(e))
```

Comment:
This looks good.

One thing you could do if you wanted, rather than repeat the exact same code twice this would make for a good *context manager* (once you've fixed the other things).  You could put it in `astropy.io.fits.util` or `.file` and call it something like `_free_space_check` so you can use it like:

```python
with _free_space_check(hdulist):
    hdulist._writeto(hdulist._file)
```

(same for the other usage further up, but the stuff you currently have in the try/except block would be what you put in the `with` block).

You can implement this like:
```python
from contextlib import contextmanager

@contextmanager
def _free_space_check(hdul):
    try:
        yield
    except IOError as exc:
        # Do the same stuff you do to handle an IOError currently
```

You can read more about `contextmanager` here: https://docs.python.org/2/library/contextlib.html#contextlib.contextmanager  This is how you can write reusable try/except patterns (among other things, though that's a common use).